LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ClientCertificateValidatorService.cs

```diff
@@ -51,8 +54,18 @@ public async Task<bool> ValidateAsync(X509Certificate2 certificate, X509Chain? c
             }
 
             // Only validation is currently done on thumprint
-            var thumbprints = await this.stationConfigurationService.GetAllowedClientThumbprintsAsync(stationEui, token);
-            return thumbprints.Any(t => t.Equals(certificate.Thumbprint, StringComparison.OrdinalIgnoreCase));
+            try
+            {
+                var thumbprints = await this.stationConfigurationService.GetAllowedClientThumbprintsAsync(stationEui, token);
+                var thumbprintFound = thumbprints.Any(t => t.Equals(certificate.Thumbprint, StringComparison.OrdinalIgnoreCase));
+                if (!thumbprintFound)
+                    this.logger.LogDebug($"'{certificate.Thumbprint}' was not found as allowed thumbprint for {stationEui}");
```

Comment:
Should we plan for a metric for when auth is refused because of certs? Could indicate a configuration issue

Reply:
I am not sure a metric could help here, but if you think it's needed, let's open another issue for tracking it 👍